python/activity_stream/activity_stream.py

```diff
@@ -0,0 +1,552 @@
+        # only once per user
+        reply_users_dup_check = []
+        for reply_user in all_reply_users:
+            
+            unique_user = (reply_user["type"], reply_user["id"]) 
+            
+            if unique_user not in reply_users_dup_check: 
+                reply_users_dup_check.append(unique_user)
+                self._data_manager.request_user_thumbnail(reply_user["type"], 
+                                                          reply_user["id"],
+                                                          reply_user["image"])
+        
+        self._bundle.log_debug("...done")
+        
+        # and now request an update check
+        self._bundle.log_debug("Ask db manager to ask shotgun for updates...")
+        self._data_manager.rescan()
+        self._bundle.log_debug("...done")
+            
+    ############################################################################
+    # internals
+        
+    def _load_stylesheet(self):
+        """
+        Loads in a stylesheet from disk
+        """
+        qss_file = os.path.join(os.path.dirname(os.path.abspath(__file__)), "style.qss")
+        try:
+            f = open(qss_file, "rt")
+            qss_data = f.read()
+            # apply to widget (and all its children)
+            self.setStyleSheet(qss_data)
+        finally:
+            f.close()
+        
+    def _clear(self):
+        """
+        Clear the widget. This will remove all items the UI
+        """
+        self._bundle.log_debug("Clearing UI...")
+        # before we begin widget operations, turn off visibility
+        # of the whole widget in order to avoid recomputes        
+        self.setVisible(False)
+        
+        # scroll to top
+        self.ui.activity_stream_scroll_area.verticalScrollBar().setValue(0)
+        
+        try:
+            self._bundle.log_debug("Clear loading widget")
+            self._clear_loading_widget()
+
+            self._bundle.log_debug("Removing all widget items")
+            for x in self._widgets.values():
+                # remove widget from layout:
+                self.ui.activity_stream_layout.removeWidget(x)
+                # set it's parent to None so that it is removed from the widget hierarchy
+                x.setParent(None)
+                # mark it to be deleted when event processing returns to the main loop
+                x.deleteLater()
+        
+            self._bundle.log_debug("Clearing python data structures")
+            self._widgets = {}
+    
+            self._bundle.log_debug("Removing expanding widget")
+            for w in self._other_widgets:
+                self.ui.activity_stream_layout.removeWidget(w)
+                w.setParent(None)
+                w.deleteLater()
+            self._other_widgets = []
+        
+        finally:
+            # make the window visible again and trigger a redraw
+            self.setVisible(True)
+            
+    def _clear_loading_widget(self):
```

Comment:
I think the reason for this is that it needs to be part of the scroll area, so that you can scroll down while it is loading. I'll add a comment to clarify.